contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -101,6 +123,324 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 	}), middleware.After)
 }
 
+func extractResourceNameFromParams(requestInput middleware.InitializeInput, awsService string) (resourceNameKey string, resourceNameValue string) {
+	switch awsService {
+	case "SQS":
+		resourceNameKey, resourceNameValue = extractQueueName(requestInput)
+	case "S3":
+		resourceNameKey, resourceNameValue = extractBucketName(requestInput)
+	case "SNS":
+		resourceNameKey, resourceNameValue = extractTopicName(requestInput)
+	case "DynamoDB":
+		resourceNameKey, resourceNameValue = extractTableName(requestInput)
+	case "Kinesis":
+		resourceNameKey, resourceNameValue = extractStreamName(requestInput)
+	case "EventBridge":
+		resourceNameKey, resourceNameValue = extractRuleName(requestInput)
+	case "SFN":
+		resourceNameKey, resourceNameValue = extractStateMachineName(requestInput)
+	}
+	return resourceNameKey, resourceNameValue
+}
+
+func extractQueueName(requestInput middleware.InitializeInput) (string, string) {
+	queueNameValue := ""
+	switch params := requestInput.Parameters.(type) {
+	case *sqs.SendMessageInput:
+		if params.QueueUrl != nil {
```

Comment:
instead of repeating `if params.QueuUrl != nil` check, can we do the check first before entering the switch statement? This we we only write it once and can skip the switch block if nil. 
The same applies to other methods too.

Reply:
You can't check before the switch statement, but you can check afterwards (I made a similar suggestion to reduce the repetition).